test/AssemblySet_basic_test.py

```diff
@@ -1,146 +1,209 @@
-    )[0]
-    assert "set_ref" in res
-    assert "set_info" in res
-    assert len(res["set_info"]) == INFO_LENGTH
-
-    assert res["set_info"][1] == set_name
-    assert "item_count" in res["set_info"][10]
-    assert res["set_info"][10]["item_count"] == str(n_items)
-
-    # test get of that object
-    d1 = set_api_client.get_assembly_set_v1(context, {"ref": ws_name + "/" + set_name})[
-        0
-    ]
-    check_empty_set(d1, set_description)
-
-    d2 = set_api_client.get_assembly_set_v1(
-        context, {"ref": res["set_ref"], "include_item_info": 1}
-    )[0]
-    check_empty_set(d2, set_description)
+        set_api_client.save_assembly_set_v1,
+        API_CLASS.set_items_type(),
+    )
+
+
+def test_save_assembly_set_no_items_list(
+    set_api_client: SetAPI, context: dict[str, str | list]
+) -> None:
+    check_save_set_no_items_list(
+        context,
+        set_api_client.save_assembly_set_v1,
+        API_CLASS.set_items_type(),
+    )
+
+
+# it's possible to create an empty assembly set, so let's test it!
+def test_save_assembly_set_no_assemblies(
+    empty_assembly_set: dict[str, Any],
+) -> None:
+    check_save_set_output(**empty_assembly_set)
+
+
+@pytest.mark.parametrize(
+    "ref_args",
+    [
+        pytest.param("__SET_REF__", id="set_ref"),
+        pytest.param("__WS_NAME__SET_NAME__", id="ws_name_set_name"),
+    ],
+)
+@pytest.mark.parametrize(
+    "get_method_args",
+    [
+        pytest.param({}, id="empty"),
+        pytest.param({INC_ITEM_INFO: 1}, id="inc_item_info"),
+        pytest.param(
+            {INC_ITEM_REF_PATHS: 1},
+            id="inc_ref_path",
+        ),
+        pytest.param(
+            {
+                INC_ITEM_INFO: 1,
+                INC_ITEM_REF_PATHS: 1,
+            },
+            id="inc_item_info_ref_path",
+        ),
+        pytest.param(
+            {INC_ITEM_INFO: 1, INC_ITEM_REF_PATHS: 1, REF_PATH_TO_SET: ["YES"]},
+            id="inc_item_info_ref_path_w_ref_path",
+        ),
+        pytest.param(
+            {
+                INC_ITEM_INFO: 0,
+                INC_ITEM_REF_PATHS: 0,
+            },
+            id="no_incs",
+        ),
+    ],
+)
+def test_get_assembly_set(
+    set_api_client: SetAPI,
+    context: dict[str, str | list],
+    ws_name: str,
+    assembly_set: dict[str, int | str | dict[str, Any]],
+    ref_args: str,
+    get_method_args: dict[str, str | int],
+    config: dict[str, Any],
+) -> None:
+    assembly_set_ref = assembly_set["obj"]["set_ref"]
+    params = {}
+    if ref_args == "__SET_REF__":
+        params["ref"] = assembly_set_ref
+    else:
+        params["ref"] = f"{ws_name}/{assembly_set['set_name']}"
+
+    for param in [INC_ITEM_INFO, INC_ITEM_REF_PATHS, REF_PATH_TO_SET]:
+        if param in get_method_args:
+            params[param] = get_method_args[param]
+
+    if params.get(REF_PATH_TO_SET, []) != []:
+        # add in a value for the REF_PATH_TO_SET
+        params[REF_PATH_TO_SET] = [assembly_set_ref]
```

Comment:
this is a bit crap but I didn't want to have to set up a whole reference chain for this test